quickwit/quickwit-indexing/src/source/queue_sources/memory_queue.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{BTreeMap, VecDeque};
+use std::fmt;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+
+use async_trait::async_trait;
+use quickwit_storage::OwnedBytes;
+use ulid::Ulid;
+
+use super::message::{MessageMetadata, RawMessage};
+use super::Queue;
+
+#[derive(Default)]
+struct InnerState {
+    in_queue: VecDeque<RawMessage>,
+    in_flight: BTreeMap<String, RawMessage>,
+    acked: Vec<RawMessage>,
+}
+
+impl fmt::Debug for InnerState {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Queue")
+            .field("in_queue_count", &self.in_queue.len())
+            .field("in_flight_count", &self.in_flight.len())
+            .field("acked_count", &self.acked.len())
+            .finish()
+    }
+}
+
+/// A simple in-memory queue
+#[derive(Clone, Default, Debug)]
+pub struct MemoryQueue {
```

Comment:
is this just for test mocking? If so can you make that clear? Not compile it on release? 
Rename it MemoryQueueForTest or whatever?

Reply:
> Not compile it on release? 

It's not compiled on release: https://github.com/quickwit-oss/quickwit/blob/37836be97f59e7420298dd6bbc668c3ba6d3327f/quickwit/quickwit-indexing/src/source/queue_sources/mod.rs#L22-L23

> Rename it MemoryQueueForTest or whatever?

:+1: